2-promises/solution.js

```diff
@@ -23,17 +23,34 @@ const id = yourRandomMethod() //third run
 6. use promise chaining accordingly, don't forget error handling
 7. log the resultant fullname, or the error, at the end
 */
+const lastnames = require('./lastnames');
+const firstnames = require('./firstnames');
+
+function getRandomId() {
+    const random = Math.random();
+    if (random < 0.5) return '❌';
+    const id = Math.floor(Math.random() * 100)
+    return id
+}
 
 function solution() {
     // YOUR SOLUTION GOES HERE
 
     // You generate your id value here
-
+    const id = getRandomId()
+    let lastname;
+    let firstname;
     // You call the lastnames method with your id
-
-    // You call the firstname method
-
-    // You log the fullname, or error, here
+    lastnames(id)
+        .then((value) => {
+            lastname = value
+            return firstnames(lastname)
+        })
+        .then((name) => {
+            firstname = name
+        })
+        .catch((error) => console.error(error))
+        .finally(() => console.log(`fullname: ${firstname} ${lastname}`))
```

Comment:
`.finally()` is usually used for clean up or actions that should happen either if the promise resolves or rejects. This statement should be in the `then` part of the code, as right now it prints `fullname: undefined undefined` when we get an error